src/main/java/zd/zero/waifu/motivator/plugin/assets/AssetManager.kt

```diff
@@ -75,11 +76,16 @@ object AssetManager {
 
     private fun resolveTheAssetUrl(localAssetPath: Path, remoteAssetUrl: String): Optional<String> =
         when {
-            hasAssetChanged(localAssetPath, remoteAssetUrl) ->
-                downloadAndGetAssetUrl(localAssetPath, remoteAssetUrl)
-            Files.exists(localAssetPath) ->
+            Files.exists(localAssetPath) -> {
+                ApplicationManager.getApplication().executeOnPooledThread {
+                    if (hasAssetChanged(localAssetPath, remoteAssetUrl)) {
+                        downloadAndGetAssetUrl(localAssetPath, remoteAssetUrl)
+                    }
+                }
+
```

Comment:
As the asset-check now prioritizes the local assets, it's now moved to only check if the asset is changed if it exists in the first place, and while at it, retrieve new assets.